tests/group_ops.rs

```diff
@@ -154,81 +169,86 @@ fn group_rotate_private_key_non_admin() -> Result<(), IronOxideErr> {
         &gen_jwt(Some(account_id.id())).0,
         USER_PASSWORD,
         &Default::default(),
-    )?;
-    let creator_sdk = ironoxide::initialize(&device)?;
+    )
+    .await?;
+    let creator_sdk = ironoxide::initialize(&device).await?;
     // making non-default groups so I can specify needs_rotation of true
-    let group_create1 = creator_sdk.group_create(&GroupCreateOpts::new(
-        None,
-        None,
-        true,
-        true,
-        None,
-        vec![],
-        vec![],
-        true,
-    ))?;
+    let group_create1 = creator_sdk
+        .group_create(&GroupCreateOpts::new(
+            None,
+            None,
+            true,
+            true,
+            None,
+            vec![],
+            vec![],
+            true,
+        ))
+        .await?;
     assert_eq!(group_create1.needs_rotation(), Some(true));
-    let group_create2 = creator_sdk.group_create(&GroupCreateOpts::new(
-        None,
-        None,
-        true,
-        true,
-        None,
-        vec![],
-        vec![],
-        true,
-    ))?;
+    let group_create2 = creator_sdk
+        .group_create(&GroupCreateOpts::new(
+            None,
+            None,
+            true,
+            true,
+            None,
+            vec![],
+            vec![],
+            true,
+        ))
+        .await?;
     assert_eq!(group_create2.needs_rotation(), Some(true));
 
-    let init_and_rotation_check = ironoxide::initialize_check_rotation(&device)?;
+    let init_and_rotation_check = ironoxide::initialize_check_rotation(&device).await?;
     let (user_result, group_result) = match init_and_rotation_check {
         InitAndRotationCheck::NoRotationNeeded(_) => {
             panic!("both user and groups should need rotation!");
         }
-        InitAndRotationCheck::RotationNeeded(io, rot) => rot.rotate_all(&io, USER_PASSWORD)?,
+        InitAndRotationCheck::RotationNeeded(io, rot) => rot.rotate_all(&io, USER_PASSWORD).await?,
     };
     assert!(user_result.is_some());
     assert!(group_result.is_some());
 
     assert_eq!(group_result.unwrap().len(), 2);
 
-    let user_result = IronOxide::user_verify(&jwt)?;
+    let user_result = IronOxide::user_verify(&jwt).await?;
     assert!(!user_result.unwrap().needs_rotation());
 
-    let group1_result = creator_sdk.group_get_metadata(group_create1.id())?;
-    let group2_result = creator_sdk.group_get_metadata(group_create2.id())?;
+    let group1_result = creator_sdk.group_get_metadata(group_create1.id()).await?;
+    let group2_result = creator_sdk.group_get_metadata(group_create2.id()).await?;
 
     assert!(!group1_result.needs_rotation().unwrap());
     assert!(!group2_result.needs_rotation().unwrap());
 
     Ok(())
 }
 
-#[test]
-fn group_get_metadata() -> Result<(), IronOxideErr> {
-    let admin_sdk = initialize_sdk()?;
-    let member_sdk = initialize_sdk()?;
-    let nonmember_sdk = initialize_sdk()?;
+#[tokio::test]
+async fn group_get_metadata() -> Result<(), IronOxideErr> {
```

Comment:
I assume that this function would be a good place to use `join`s instead of back-to-back `await`s. This might not be the PR to do so, but this function in particular looks like we could speed it up.

Reply:
Fair observation. I'm going for a 1-1 conversion in this PR, however and these were being done in serial before.